### DISCUSSION TITLE
Stellarium 1.1 & master (2022-11-23): artifact pattern in the background

### DISCUSSION THREAD

**Discussion by dhdeangelis** [Stellarium/stellarium#2869] [General]
I have noticed that both Stellarium 1.1 and master code downloaded and built 2022-11-23 show a strange pattern of artifacts in the background as well as vertically "smeared" stars:

![image](https://user-images.githubusercontent.com/51515911/203730536-428f4166-430d-4ad7-8d55-e24da2344aed.png)

I have never seen this before with Stellarium. I am *not* stating that this is a problem with Stellarium. I am trying to find the cause and envisage a solution.

I use OpenSUSE Tumbleweed which is a rolling release and has a very high rate of update. Recently Mesa (now at version 22.2.4-333.1) has been updated and this has possibly contributed to this. 

What could be going on here? What is important to consider with this type of problem?

Thanks in advance

**Comment by 10110111** [comment]
This may be a bug in the GPU drivers. You say it's Mesa, and I suppose you may be using an NVIDIA GPU. In this case, try using the proprietary `nvidia` driver, since `nouveau` is usually unreliable and prone to artifacts.

**Reply by dhdeangelis** [reply]
Thank you, installing the proprietary NVIDIA driver solved the problem.

**Comment by Reue** [comment]
I'm using Debian Testing and Stellarium last version (25.4). I'm having the exact same problem (artifacts) as user dhdeangelis. And I'm also using the open-source driver (nouveau) on a system whose video card GPU is Nvidia.

So, after more than 3 years, the problem remains...

Installing Nvidia's proprietary drivers is not a solution, as it would imply crucial breaches of privacy and security in systems that, for some reason, are exclusively based on free and open-source software (I'm using Debian for a reason).
On the other hand, it wouldn't make any sense to introduce proprietary drivers into the system because of a single software (Stellarium).

Furthermore, it's worth mentioning that I'm using other astronomy programs, such as Celestia and Kstars, and both work without any problems with the nouveau driver...


Therefore, I ask the developers to resolve this very serious bug, because without a proper solution Stellarium is completely unusable. Thanks in advance and a Happy New Year!

**Reply by gzotti** [reply]
Solve it, and receive a free coffee.

**Reply by 10110111** [reply]
Unfortunately, not all free software is equally good. Nouveau has always been too buggy to be usable on most GPUs, so anytime you see graphics artifacts with this driver but not the others, you can safely bet that the bug is on the driver's side. If you crucially depend on having nouveau instead of the official proprietary driver, you have to report all the bugs to Mesa.

While we could add some workarounds to Stellarium (which also need to be developed, and the problem itself reproduced), I wouldn't really vote for doing so, simply because nouveau has been a can of bugs for all the lifetime of the project, and we'd have to add many more workarounds for other issues that people have been solving by switching to the driver that works.

**Reply by 10110111** [reply]
BTW, just realized that you might be able to work around this particular issue by disabling dithering (<kbd>F2</kbd> → _Tools_ → _Dithering_ = _None_). But generally, my previous message about bug reports to Mesa is still valid.

**Reply by dhdeangelis** [reply]
> BTW, just realized that you might be able to work around this particular issue by disabling dithering (F2 → _Tools_ → _Dithering_ = _None_). But generally, my previous message about bug reports to Mesa is still valid.

Thanks for this recommendation. Setting dithering to none allows the screen to be legible even when using a bad/buggy Nouveau open driver.

I also believe there is no reason for Stellarium devteam to do anything on this regard. The problem lies elsewhere. I am closing this discussion.
